### PR TITLE
Pre searching

### PR DESCRIPTION
**Pull request by ngailapdi**

### CODE REVIEW

File: static/templyte/src/app/my-templates/my-templates.component.html [hunk 1]
```diff
@@ -25,6 +25,9 @@
     <mat-card class="template-card">
         <p class="template-title">{{template.templateName}}</p>
         <p>by {{template.authorName}}</p>
+        <p>{{template.templateDescription}}</p>
+        <p>Creation date: {{template.templateCreateDate}}</p>
```
Comment (debkbanerji) [inline static/templyte/src/app/my-templates/my-templates.component.html]: Does this display properly?

File: static/templyte/src/app/my-templates/my-templates.component.html [hunk 2]
```diff
@@ -25,6 +25,9 @@
...
+        <p>Number of downloads: {{template.templateNumDownload}}</p>
         <mat-chip-list>
```
Comment (debkbanerji) [inline static/templyte/src/app/my-templates/my-templates.component.html]: Should last downloaded date be displayed?

File: static/templyte/src/app/download-template/download-template.component.ts [hunk 3]
```diff
@@ -76,7 +74,7 @@ export class DownloadTemplateComponent implements OnInit {
         component.validateEnteredVariables();
         component.templateRenderInfoRef.snapshotChanges().subscribe(data => {
             const payload_val = data.payload.val();
-            const fileEndings = payload_val.fileEndings;
+            const fileEndings = payload_val.fileEndings;
```
Comment (debkbanerji) [inline static/templyte/src/app/download-template/download-template.component.ts]: Why the change?

File: static/templyte/src/app/download-template/download-template.component.ts [hunk 4]
```diff
@@ -97,12 +95,12 @@ export class DownloadTemplateComponent implements OnInit {
...
-                component.templateDirectoryInfoDatabaseRef.child('/templateNumDownload').once('value', function(snapshot) {
-                    component.templateDirectoryInfoRef.update({
-                        'templateNumDownload': snapshot.val() + 1,
-                        'templateLastDownloadDate': formatDate(new Date(), 'yyyy/MM/dd', 'en')
-                    })
+                component.templateDirectoryInfoDatabaseRef.child('/templateNumDownload').transaction(function(snapshot) {
+                    return snapshot+1;
                 });
+                component.templateDirectoryInfoRef.update({
+                    'templateLastDownloadDate': formatDate(new Date(), 'yyyy/MM/dd', 'en')});
```
Comment (debkbanerji) [inline static/templyte/src/app/download-template/download-template.component.ts]: Be consistent with what elements appear on what lines.

File: static/templyte/src/app/download-template/download-template.component.ts [hunk 5]
```diff
@@ -90,9 +95,28 @@ export class DownloadTemplateComponent implements OnInit {
...
         });
+        
+
+    }
+    update(callback) {
+        const component = this;
+
+    }
+
+    getUpdateValue(callback) {
```
Comment (debkbanerji) [inline static/templyte/src/app/download-template/download-template.component.ts]: What does this do?

File: static/templyte/src/app/download-template/download-template.component.ts [hunk 6]
```diff
@@ -90,9 +95,28 @@ export class DownloadTemplateComponent implements OnInit {
                     'cancelable': false
                 });
                 linkElement.dispatchEvent(clickEvent);
+                component.templateDirectoryInfoDatabaseRef.child('/templateNumDownload').transaction(function(snapshot) {
+                    return snapshot+1;
```
Comment (debkbanerji) [inline static/templyte/src/app/download-template/download-template.component.ts]: Be careful of spacing - should be consistent

File: static/templyte/src/app/download-template/download-template.component.ts [hunk 7]
```diff
@@ -90,9 +95,28 @@ export class DownloadTemplateComponent implements OnInit {
...
+                });
+                component.templateDirectoryInfoRef.update({
+                    'templateLastDownloadDate': formatDate(new Date(), 'yyyy/MM/dd', 'en')});
```
Comment (debkbanerji) [inline static/templyte/src/app/download-template/download-template.component.ts]: Why are we storing dates as strings? - We're not going to be able to index on these correctly.

Comment (debkbanerji) [inline static/templyte/src/app/download-template/download-template.component.ts]: Why are we using 'update' instead of 'set' - this way you're doing more complicated logic

File: static/templyte/src/app/download-template/download-template.component.ts [hunk 8]
```diff
@@ -90,9 +95,28 @@ export class DownloadTemplateComponent implements OnInit {
...
+
             });
 
         });
+        
+
```
Comment (debkbanerji) [inline static/templyte/src/app/download-template/download-template.component.ts]: Weird spacing

File: static/templyte/src/app/create-template/create-template.component.html [hunk 9]
```diff
@@ -22,6 +22,8 @@
     <div class="container">
         <input class="form-control" id="template-name-input" matInput [(ngModel)]="templateName"
                placeholder="What is the name of your template?" value="My Awesome Template!">
+        <input class="form-control" id="template-description-input" matInput [(ngModel)]="templateDescription"
```
[the remark this answers is not in the document]
Comment (debkbanerji) [reply static/templyte/src/app/create-template/create-template.component.html]: Comment still applies

File: static/templyte/src/app/download-template/download-template.component.ts [hunk 10]
```diff
@@ -90,9 +97,28 @@ export class DownloadTemplateComponent implements OnInit {
...
+                        'templateLastDownloadDate': formatDate(new Date(), 'yyyy/MM/dd', 'en')
+                    })
+                });
             });
 
         });
+        
+
+    }
+    update(callback) {
```
[the remark this answers is not in the document]
Comment (ngailapdi) [reply static/templyte/src/app/download-template/download-template.component.ts]: deleted

File: static/templyte/src/app/download-template/download-template.component.ts [hunk 11]
```diff
@@ -5,6 +5,8 @@ import {Component, NgZone, OnInit} from '@angular/core';
 import {AngularFireDatabase, AngularFireObject} from 'angularfire2/database';
 import {Observable} from 'rxjs';
 import {ApiInterfaceService} from '../providers/api-interface.service';
+import * as firebase from 'firebase';
+import { Reference } from 'firebase/database';
```
Comment (debkbanerji) [inline static/templyte/src/app/download-template/download-template.component.ts]: Make spacing consistent
